docs/proposals/01-extensibility.md

```diff
@@ -0,0 +1,844 @@
+    message: DNS record has been created and is available.
+    reason: RecordResolvable
+    status: "True"
+    type: Available
+  providerStatus:
+    apiVersion: aws.extensions.gardener.cloud/v1alpha1
+    kind: DNSStatus
+    ...
+```
+
+##### Infrastructure provisioning
+
+The infrastructure contains the information about VPC, networks, security groups, availability zones, ..., basically, everything that needs to be prepared before actual VMs/load balancers/... can be provisioned.
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: Infrastructure
+metadata:
+  name: infrastructure
+  namespace: shoot--core--aws-01
+spec:
+  type: aws
+  providerConfig:
+    apiVersion: aws.extensions.gardener.cloud/v1alpha1
+    kind: InfrastructureConfig
+    networks:
+      vpc:
+        cidr: 10.250.0.0/16
+      internal:
+      - 10.250.112.0/22
+      public:
+      - 10.250.96.0/22
+      workers:
+      - 10.250.0.0/19
+    zones:
+    - eu-west-1a
+  dns:
+    apiserver: api.aws-01.core.example.com
+  region: eu-west-1
+  secretRef:
+    name: my-aws-credentials
+  sshPublicKey: |
+    base64(key)
+status:
+  observedGeneration: ...
+  state: ...
+  lastError: ..
+  lastOperation: ...
+  providerStatus:
+    apiVersion: aws.extensions.gardener.cloud/v1alpha1
+    kind: InfrastructureStatus
+    vpc:
+      id: vpc-1234
+      subnets:
+      - id: subnet-acbd1234
+        name: workers
+        zone: eu-west-1
+      securityGroups:
+      - id: sg-xyz12345
+        name: workers
+    iam:
+      nodesRoleARN: <some-arn>
+      instanceProfileName: foo
+    ec2:
+      keyName: bar
+```
+
+##### Backup infrastructure provisioning
+
+The `BackupInfrastructure` CRD in the Seeds tells the cloud-specific controller to prepare a blob store bucket/container which can later be used to store etcd backups.
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: BackupInfrastructure
+metadata:
+  name: etcd-backup
+  namespace: shoot--core--aws-01
+spec:
+  type: aws
+  region: eu-west-1
+  storageContainerName: asdasjndasd-1293912378a-2213
+  secretRef:
+    name: my-aws-credentials
+status:
+  observedGeneration: ...
+  state: ...
+  lastError: ..
+  lastOperation: ...
+```
+
+##### Cloud config (user-data) for bootstrapping machines
+
+Gardener keeps keeping knowledge about the content of the cloud config scripts, but it will hand over it to the respective OS-specific controller which will generate the specific valid representation.
+Gardener creates two `MachineCloudConfig` CRDs, one for the cloud-config-downloader (which will later flow into the `WorkerPool` CRD) and one for the real cloud-config (which will be stored as a `Secret` in the Shoot's `kube-system` namespace, and downloaded and executed from the cloud-config-downloader on the machines).
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: MachineCloudConfig
+metadata:
+  name: pool-01-downloader
```

Comment:
Agreed, we want to do the same since long and discussed it months ago when @hardikdr and @prashanth26 were visiting us on-site, however, we found it not trivial to achieve and postponed it. We still want to do it, no doubts, but maybe let's keep it outside of this proposal and tackle it separately.